### DISCUSSION TITLE
(Proposal) Pinned center keyframe manipulation

### DISCUSSION THREAD

**Discussion by psychoanima** [opentoonz/opentoonz#6495] [Ideas]
Dear developers, 

since pinned center keyframes are now available on xsheet it helped me a lot to understand how pinned center is really working with IK. 

Now I am presenting you video mocup / proposal how to make this pinned center useful for any IK animation that needs re-timing.

https://1drv.ms/v/s!AtMv1SrHrjT3qx8ndo61PaK4tFxT

In my example I want to re-time one part of my character walk. 
When you stretch bone keyframes, pinned center keyframe is staying at original place - it's not following the bone keyframes which is causing a problem showed at the beginning of my video. To correct the problem you need to make several steps such as catching old pinned bone, adding pin to it, removing extra keyframes, adding pin to corrected bone... (see video above).

I was thinking a lot what would be the best and simplest way to re-pin and not cause the headache to developers and animators. Making Pinned keyframe to follow regular keyframes is not an option. Who knows how much of rocket science is necessary to make that work.

That's why I am proposing simple feature: drag and drop option (second part of my video). When animator re-time his IK animation, the only thing that is left to do is to drag pinned center keyframe to correct spot. And that's basically it. That simple. Waiting for your opinion. 

Thanks for your time, cut-out animators loves ya all!

**Comment by ghost** [comment]
Based on the discussion in #1207 feature requests are going to be closed here on GitHub unless there is a developer actively working on the feature. Feature requests can be discussed at the OpenToonz Google Group: https://groups.google.com/forum/#!forum/opentoonz_en

**Comment by RodneyBaker** [comment]
Reopening as feature requests are now allowed and encouraged for Opentoonz github.

@psychoanima Your referenced video link is no longer valid.
